src/ucp/core/ucp_context.c

```diff
@@ -138,6 +138,13 @@ const size_t ucp_context_est_bcopy_bw[UCS_CPU_VENDOR_LAST] = {
     [UCS_CPU_VENDOR_ZHAOXIN]     = UCP_CPU_EST_BCOPY_BW_DEFAULT
 };
 
+static const char *ucp_reg_nb_mem_type_names[] = {
+    [UCP_MEMTYPE_REG_NB_TYPES_AUTO] = "auto",
+    [UCP_MEMTYPE_REG_NB_TYPES_HOST] = "host",
+    [UCP_MEMTYPE_REG_NB_TYPES_CUDA] = "cuda",
+    [UCP_MEMTYPE_REG_NB_TYPES_LAST] = NULL
+};
```

Comment:
i think we can reuse `ucs_memory_type_names` (like `MEMTYPE_REG_WHOLE_ALLOC_TYPES` does) and use `UCS_MEMORY_TYPE_LAST` to indicate auto value.

Reply:
I think we can define special config value for UCS_CONFIG_TYPE_BITMAP case, "all", which would set the bitmap to 0xff..ff.
"all" would mean to use any supported (by UCT) memory type.

Reply:
sounds good, but we would need to be able to handle `auto` as well, so looks like need a bit for this (maybe use `UNKNOWN`)

Reply:
what would be difference between "auto" and "all"?

Reply:
we can apply different default values on different platforms. E.g. sometimes it will be 0

Reply:
sounds weird to have specific platform logic in UCP
anyway I think "auto" is not needed at this point. we may consider adding it later.